### PR TITLE
Fix unhandled compiler exceptions being not result in test failures

### LINKED ISSUE #34553: Complier throws exception for invalid floating points like `12.1e`, but the parserTestRunner does not fail as the exception is handled internally
**Description:**
`float floatResult = 12.1e;` crashes with following stacktrace,

<details>
  <summary><b>stacktrace</b>  (<i>click to expand</i>)</summary>

```
[2022-01-07 11:31:59,974] SEVERE {b7a.log.crash} - For input string: "12.1e" 
java.lang.NumberFormatException: For input string: "12.1e"
	at java.base/jdk.internal.math.FloatingDecimal.readJavaFormatString(FloatingDecimal.java:2054)
	at java.base/jdk.internal.math.FloatingDecimal.parseDouble(FloatingDecimal.java:110)
	at java.base/java.lang.Double.parseDouble(Double.java:543)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.Types.validateFloatLiteral(Types.java:3587)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.getTypeOfDecimalFloatingPointLiteral(TypeChecker.java:630)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.setLiteralValueAndGetType(TypeChecker.java:702)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.visit(TypeChecker.java:425)
	at org.wso2.ballerinalang.compiler.tree.expressions.BLangLiteral.accept(BLangLiteral.java:55)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.checkExpr(TypeChecker.java:380)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.checkExpr(TypeChecker.java:354)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.visit(SemanticAnalyzer.java:967)
	at org.wso2.ballerinalang.compiler.tree.BLangSimpleVariable.accept(BLangSimpleVariable.java:54)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeNode(SemanticAnalyzer.java:3920)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeNode(SemanticAnalyzer.java:3891)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeDef(SemanticAnalyzer.java:3883)
	at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.visit(SemanticAnalyzer.java:341)
	at org.wso2.ballerinalang.compiler.tree.BLangPackage.accept(BLangPackage.java:167)
```
</details>

<details>
  <summary><b>ParserTestRunner log for affected test cases</b>  (<i>click to expand</i>)</summary>

```
ballerina-new-parser-test-suite > jballerina-test > org.ballerinalang.test.ParserTestRunner > test[834](method_call_expr_source_03.bal, /home/lochanaj/Ballerina/ballerina-lang/compiler/ballerina-parser/src/test/resources/expressions/access-expr/method_call_expr_source_03.bal) STANDARD_ERROR
    ballerina: Oh no, something really went wrong. Bad. Sad.

    We appreciate it if you can report the code that broke Ballerina in
    https://github.com/ballerina-platform/ballerina-lang/issues with the
    log you get below and your sample code.

    We thank you for helping make us better.

    [2022-01-07 11:10:29,392] SEVERE {b7a.log.crash} - For input string: "12.E" 
    java.lang.NumberFormatException: For input string: "12.E"
        at java.base/jdk.internal.math.FloatingDecimal.readJavaFormatString(FloatingDecimal.java:2054)
        at java.base/jdk.internal.math.FloatingDecimal.parseDouble(FloatingDecimal.java:110)
        at java.base/java.lang.Double.parseDouble(Double.java:543)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.Types.validateFloatLiteral(Types.java:3587)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.getTypeOfDecimalFloatingPointLiteral(TypeChecker.java:657)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.setLiteralValueAndGetType(TypeChecker.java:702)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.visit(TypeChecker.java:425)
        at org.wso2.ballerinalang.compiler.tree.expressions.BLangLiteral.accept(BLangLiteral.java:55)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.checkExpr(TypeChecker.java:380)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.checkExpr(TypeChecker.java:354)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.checkExpr(TypeChecker.java:350)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.visit(TypeChecker.java:4333)
        at org.wso2.ballerinalang.compiler.tree.expressions.BLangBinaryExpr.accept(BLangBinaryExpr.java:67)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.checkExpr(TypeChecker.java:380)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.checkExpr(TypeChecker.java:354)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.handleDeclaredWithVar(SemanticAnalyzer.java:1454)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.visit(SemanticAnalyzer.java:879)
        at org.wso2.ballerinalang.compiler.tree.BLangSimpleVariable.accept(BLangSimpleVariable.java:54)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeNode(SemanticAnalyzer.java:3920)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeNode(SemanticAnalyzer.java:3891)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeDef(SemanticAnalyzer.java:3883)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.visit(SemanticAnalyzer.java:1778)
        at org.wso2.ballerinalang.compiler.tree.statements.BLangSimpleVariableDef.accept(BLangSimpleVariableDef.java:50)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeNode(SemanticAnalyzer.java:3920)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeNode(SemanticAnalyzer.java:3891)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeStmt(SemanticAnalyzer.java:3887)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.visit(SemanticAnalyzer.java:472)
        at org.wso2.ballerinalang.compiler.tree.BLangBlockFunctionBody.accept(BLangBlockFunctionBody.java:60)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeNode(SemanticAnalyzer.java:3920)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.visit(SemanticAnalyzer.java:455)
        at org.wso2.ballerinalang.compiler.tree.BLangFunction.accept(BLangFunction.java:76)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeNode(SemanticAnalyzer.java:3920)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeNode(SemanticAnalyzer.java:3891)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeDef(SemanticAnalyzer.java:3883)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.visit(SemanticAnalyzer.java:341)
        at org.wso2.ballerinalang.compiler.tree.BLangPackage.accept(BLangPackage.java:167)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyze(SemanticAnalyzer.java:303)
        at io.ballerina.projects.internal.CompilerPhaseRunner.typeCheck(CompilerPhaseRunner.java:188)
        at io.ballerina.projects.internal.CompilerPhaseRunner.performTypeCheckPhases(CompilerPhaseRunner.java:114)
        at io.ballerina.projects.ModuleContext.compileInternal(ModuleContext.java:436)
        at io.ballerina.projects.ModuleCompilationState$1.compile(ModuleCompilationState.java:45)
        at io.ballerina.projects.ModuleContext.compile(ModuleContext.java:386)
        at io.ballerina.projects.PackageCompilation.compileModulesInternal(PackageCompilation.java:213)
        at io.ballerina.projects.PackageCompilation.compileModules(PackageCompilation.java:198)
        at io.ballerina.projects.PackageCompilation.compile(PackageCompilation.java:114)
        at io.ballerina.projects.PackageCompilation.from(PackageCompilation.java:103)
        at io.ballerina.projects.PackageContext.getPackageCompilation(PackageContext.java:214)
        at io.ballerina.projects.Package.getCompilation(Package.java:141)
        at org.ballerinalang.test.BCompileUtil.jBallerinaBackend(BCompileUtil.java:187)
        at org.ballerinalang.test.BCompileUtil.compile(BCompileUtil.java:81)
        at org.ballerinalang.test.ParserTestRunner.test(ParserTestRunner.java:42)
        at jdk.internal.reflect.GeneratedMethodAccessor20.invoke(Unknown Source)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
        at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:54)
        at org.testng.internal.InvokeMethodRunnable.run(InvokeMethodRunnable.java:44)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:829)
 

ballerina-new-parser-test-suite > jballerina-test > org.ballerinalang.test.ParserTestRunner > test[844](method_call_expr_source_05.bal, /home/lochanaj/Ballerina/ballerina-lang/compiler/ballerina-parser/src/test/resources/expressions/access-expr/method_call_expr_source_05.bal) STANDARD_ERROR
    ballerina: Oh no, something really went wrong. Bad. Sad.

    We appreciate it if you can report the code that broke Ballerina in
    https://github.com/ballerina-platform/ballerina-lang/issues with the
    log you get below and your sample code.

    We thank you for helping make us better.

    [2022-01-07 11:10:29,733] SEVERE {b7a.log.crash} - For input string: "0x12.Ep" 
    java.lang.NumberFormatException: For input string: "0x12.Ep"
        at java.base/jdk.internal.math.FloatingDecimal.parseHexString(FloatingDecimal.java:2082)
        at java.base/jdk.internal.math.FloatingDecimal.readJavaFormatString(FloatingDecimal.java:1870)
        at java.base/jdk.internal.math.FloatingDecimal.parseDouble(FloatingDecimal.java:110)
        at java.base/java.lang.Double.parseDouble(Double.java:543)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.Types.validateFloatLiteral(Types.java:3587)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.getTypeOfHexFloatingPointLiteral(TypeChecker.java:666)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.setLiteralValueAndGetType(TypeChecker.java:705)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.visit(TypeChecker.java:425)
        at org.wso2.ballerinalang.compiler.tree.expressions.BLangLiteral.accept(BLangLiteral.java:55)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.checkExpr(TypeChecker.java:380)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.checkExpr(TypeChecker.java:354)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.checkExpr(TypeChecker.java:350)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.visit(TypeChecker.java:4333)
        at org.wso2.ballerinalang.compiler.tree.expressions.BLangBinaryExpr.accept(BLangBinaryExpr.java:67)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.checkExpr(TypeChecker.java:380)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.checkExpr(TypeChecker.java:354)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.handleDeclaredWithVar(SemanticAnalyzer.java:1454)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.visit(SemanticAnalyzer.java:879)
        at org.wso2.ballerinalang.compiler.tree.BLangSimpleVariable.accept(BLangSimpleVariable.java:54)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeNode(SemanticAnalyzer.java:3920)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeNode(SemanticAnalyzer.java:3891)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeDef(SemanticAnalyzer.java:3883)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.visit(SemanticAnalyzer.java:1778)
        at org.wso2.ballerinalang.compiler.tree.statements.BLangSimpleVariableDef.accept(BLangSimpleVariableDef.java:50)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeNode(SemanticAnalyzer.java:3920)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeNode(SemanticAnalyzer.java:3891)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeStmt(SemanticAnalyzer.java:3887)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.visit(SemanticAnalyzer.java:472)
        at org.wso2.ballerinalang.compiler.tree.BLangBlockFunctionBody.accept(BLangBlockFunctionBody.java:60)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeNode(SemanticAnalyzer.java:3920)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.visit(SemanticAnalyzer.java:455)
        at org.wso2.ballerinalang.compiler.tree.BLangFunction.accept(BLangFunction.java:76)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeNode(SemanticAnalyzer.java:3920)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeNode(SemanticAnalyzer.java:3891)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeDef(SemanticAnalyzer.java:3883)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.visit(SemanticAnalyzer.java:341)
        at org.wso2.ballerinalang.compiler.tree.BLangPackage.accept(BLangPackage.java:167)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyze(SemanticAnalyzer.java:303)
        at io.ballerina.projects.internal.CompilerPhaseRunner.typeCheck(CompilerPhaseRunner.java:188)
        at io.ballerina.projects.internal.CompilerPhaseRunner.performTypeCheckPhases(CompilerPhaseRunner.java:114)
        at io.ballerina.projects.ModuleContext.compileInternal(ModuleContext.java:436)
        at io.ballerina.projects.ModuleCompilationState$1.compile(ModuleCompilationState.java:45)
        at io.ballerina.projects.ModuleContext.compile(ModuleContext.java:386)
        at io.ballerina.projects.PackageCompilation.compileModulesInternal(PackageCompilation.java:213)
        at io.ballerina.projects.PackageCompilation.compileModules(PackageCompilation.java:198)
        at io.ballerina.projects.PackageCompilation.compile(PackageCompilation.java:114)
        at io.ballerina.projects.PackageCompilation.from(PackageCompilation.java:103)
        at io.ballerina.projects.PackageContext.getPackageCompilation(PackageContext.java:214)
        at io.ballerina.projects.Package.getCompilation(Package.java:141)
        at org.ballerinalang.test.BCompileUtil.jBallerinaBackend(BCompileUtil.java:187)
        at org.ballerinalang.test.BCompileUtil.compile(BCompileUtil.java:81)
        at org.ballerinalang.test.ParserTestRunner.test(ParserTestRunner.java:42)
        at jdk.internal.reflect.GeneratedMethodAccessor20.invoke(Unknown Source)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
        at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:54)
        at org.testng.internal.InvokeMethodRunnable.run(InvokeMethodRunnable.java:44)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:829)
 

ballerina-new-parser-test-suite > jballerina-test > org.ballerinalang.test.ParserTestRunner > test[914](float_literal_source_08.bal, /home/lochanaj/Ballerina/ballerina-lang/compiler/ballerina-parser/src/test/resources/expressions/basic-literals/float_literal_source_08.bal) STANDARD_ERROR
    ballerina: Oh no, something really went wrong. Bad. Sad.

    We appreciate it if you can report the code that broke Ballerina in
    https://github.com/ballerina-platform/ballerina-lang/issues with the
    log you get below and your sample code.

    We thank you for helping make us better.

    [2022-01-07 11:10:31,732] SEVERE {b7a.log.crash} - For input string: "12.e" 
    java.lang.NumberFormatException: For input string: "12.e"
        at java.base/jdk.internal.math.FloatingDecimal.readJavaFormatString(FloatingDecimal.java:2054)
        at java.base/jdk.internal.math.FloatingDecimal.parseDouble(FloatingDecimal.java:110)
        at java.base/java.lang.Double.parseDouble(Double.java:543)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.Types.validateFloatLiteral(Types.java:3587)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.getTypeOfDecimalFloatingPointLiteral(TypeChecker.java:657)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.setLiteralValueAndGetType(TypeChecker.java:702)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.visit(TypeChecker.java:425)
        at org.wso2.ballerinalang.compiler.tree.expressions.BLangLiteral.accept(BLangLiteral.java:55)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.checkExpr(TypeChecker.java:380)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.checkExpr(TypeChecker.java:354)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.visit(SemanticAnalyzer.java:1892)
        at org.wso2.ballerinalang.compiler.tree.statements.BLangAssignment.accept(BLangAssignment.java:81)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeNode(SemanticAnalyzer.java:3920)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeNode(SemanticAnalyzer.java:3891)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeStmt(SemanticAnalyzer.java:3887)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.visit(SemanticAnalyzer.java:472)
        at org.wso2.ballerinalang.compiler.tree.BLangBlockFunctionBody.accept(BLangBlockFunctionBody.java:60)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeNode(SemanticAnalyzer.java:3920)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.visit(SemanticAnalyzer.java:455)
        at org.wso2.ballerinalang.compiler.tree.BLangFunction.accept(BLangFunction.java:76)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeNode(SemanticAnalyzer.java:3920)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeNode(SemanticAnalyzer.java:3891)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeDef(SemanticAnalyzer.java:3883)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.visit(SemanticAnalyzer.java:341)
        at org.wso2.ballerinalang.compiler.tree.BLangPackage.accept(BLangPackage.java:167)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyze(SemanticAnalyzer.java:303)
        at io.ballerina.projects.internal.CompilerPhaseRunner.typeCheck(CompilerPhaseRunner.java:188)
        at io.ballerina.projects.internal.CompilerPhaseRunner.performTypeCheckPhases(CompilerPhaseRunner.java:114)
        at io.ballerina.projects.ModuleContext.compileInternal(ModuleContext.java:436)
        at io.ballerina.projects.ModuleCompilationState$1.compile(ModuleCompilationState.java:45)
        at io.ballerina.projects.ModuleContext.compile(ModuleContext.java:386)
        at io.ballerina.projects.PackageCompilation.compileModulesInternal(PackageCompilation.java:213)
        at io.ballerina.projects.PackageCompilation.compileModules(PackageCompilation.java:198)
        at io.ballerina.projects.PackageCompilation.compile(PackageCompilation.java:114)
        at io.ballerina.projects.PackageCompilation.from(PackageCompilation.java:103)
        at io.ballerina.projects.PackageContext.getPackageCompilation(PackageContext.java:214)
        at io.ballerina.projects.Package.getCompilation(Package.java:141)
        at org.ballerinalang.test.BCompileUtil.jBallerinaBackend(BCompileUtil.java:187)
        at org.ballerinalang.test.BCompileUtil.compile(BCompileUtil.java:81)
        at org.ballerinalang.test.ParserTestRunner.test(ParserTestRunner.java:42)
        at jdk.internal.reflect.GeneratedMethodAccessor20.invoke(Unknown Source)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
        at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:54)
        at org.testng.internal.InvokeMethodRunnable.run(InvokeMethodRunnable.java:44)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:829)
 

ballerina-new-parser-test-suite > jballerina-test > org.ballerinalang.test.ParserTestRunner > test[930](float_literal_source_07.bal, /home/lochanaj/Ballerina/ballerina-lang/compiler/ballerina-parser/src/test/resources/expressions/basic-literals/float_literal_source_07.bal) STANDARD_ERROR
    ballerina: Oh no, something really went wrong. Bad. Sad.

    We appreciate it if you can report the code that broke Ballerina in
    https://github.com/ballerina-platform/ballerina-lang/issues with the
    log you get below and your sample code.

    We thank you for helping make us better.

    [2022-01-07 11:10:31,902] SEVERE {b7a.log.crash} - For input string: "0x.p45" 
    java.lang.NumberFormatException: For input string: "0x.p45"
        at java.base/jdk.internal.math.FloatingDecimal.parseHexString(FloatingDecimal.java:2082)
        at java.base/jdk.internal.math.FloatingDecimal.readJavaFormatString(FloatingDecimal.java:1870)
        at java.base/jdk.internal.math.FloatingDecimal.parseDouble(FloatingDecimal.java:110)
        at java.base/java.lang.Double.parseDouble(Double.java:543)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.Types.validateFloatLiteral(Types.java:3587)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.getTypeOfHexFloatingPointLiteral(TypeChecker.java:666)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.setLiteralValueAndGetType(TypeChecker.java:705)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.visit(TypeChecker.java:425)
        at org.wso2.ballerinalang.compiler.tree.expressions.BLangLiteral.accept(BLangLiteral.java:55)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.checkExpr(TypeChecker.java:380)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.TypeChecker.checkExpr(TypeChecker.java:354)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.visit(SemanticAnalyzer.java:1892)
        at org.wso2.ballerinalang.compiler.tree.statements.BLangAssignment.accept(BLangAssignment.java:81)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeNode(SemanticAnalyzer.java:3920)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeNode(SemanticAnalyzer.java:3891)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeStmt(SemanticAnalyzer.java:3887)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.visit(SemanticAnalyzer.java:472)
        at org.wso2.ballerinalang.compiler.tree.BLangBlockFunctionBody.accept(BLangBlockFunctionBody.java:60)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeNode(SemanticAnalyzer.java:3920)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.visit(SemanticAnalyzer.java:455)
        at org.wso2.ballerinalang.compiler.tree.BLangFunction.accept(BLangFunction.java:76)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeNode(SemanticAnalyzer.java:3920)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeNode(SemanticAnalyzer.java:3891)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyzeDef(SemanticAnalyzer.java:3883)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.visit(SemanticAnalyzer.java:341)
        at org.wso2.ballerinalang.compiler.tree.BLangPackage.accept(BLangPackage.java:167)
        at org.wso2.ballerinalang.compiler.semantics.analyzer.SemanticAnalyzer.analyze(SemanticAnalyzer.java:303)
        at io.ballerina.projects.internal.CompilerPhaseRunner.typeCheck(CompilerPhaseRunner.java:188)
        at io.ballerina.projects.internal.CompilerPhaseRunner.performTypeCheckPhases(CompilerPhaseRunner.java:114)
        at io.ballerina.projects.ModuleContext.compileInternal(ModuleContext.java:436)
        at io.ballerina.projects.ModuleCompilationState$1.compile(ModuleCompilationState.java:45)
        at io.ballerina.projects.ModuleContext.compile(ModuleContext.java:386)
        at io.ballerina.projects.PackageCompilation.compileModulesInternal(PackageCompilation.java:213)
        at io.ballerina.projects.PackageCompilation.compileModules(PackageCompilation.java:198)
        at io.ballerina.projects.PackageCompilation.compile(PackageCompilation.java:114)
        at io.ballerina.projects.PackageCompilation.from(PackageCompilation.java:103)
        at io.ballerina.projects.PackageContext.getPackageCompilation(PackageContext.java:214)
        at io.ballerina.projects.Package.getCompilation(Package.java:141)
        at org.ballerinalang.test.BCompileUtil.jBallerinaBackend(BCompileUtil.java:187)
        at org.ballerinalang.test.BCompileUtil.compile(BCompileUtil.java:81)
        at org.ballerinalang.test.ParserTestRunner.test(ParserTestRunner.java:42)
        at jdk.internal.reflect.GeneratedMethodAccessor20.invoke(Unknown Source)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
        at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:54)
        at org.testng.internal.InvokeMethodRunnable.run(InvokeMethodRunnable.java:44)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:829)
```
</details>

### PR DESCRIPTION
**Pull request by lochana-chathura**
## Purpose
$subject.

Fixes #34553 

## Approach
N/A

## Samples
N/A

## Remarks
N/A

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/ModuleContext.java [hunk 1]
```diff
@@ -435,6 +435,8 @@ static void compileInternal(ModuleContext moduleContext, CompilerContext compile
             packageCache.putSymbol(pkgNode.packageID, pkgNode.symbol);
             compilerPhaseRunner.performTypeCheckPhases(pkgNode);
         } catch (Throwable t) {
+            assert false : "Compilation failed due to" +
```
Comment (pcnfernando) [inline compiler/ballerina-lang/src/main/java/io/ballerina/projects/ModuleContext.java:438]: How is -ea is enabled?
Also, can there be any case where the diagnostics will be checked after an assertion is thrown?

Comment (lochana-chathura) [reply compiler/ballerina-lang/src/main/java/io/ballerina/projects/ModuleContext.java:438]: AFAIK, with TestNG java assertions are enabled by default whereas in the production code assertions are disabled by default.  
You can try running this,
```java
    @Test
    public void testJavaAssert() {
        assert false;
    }
```
If AssertionError is thrown, no way that lines after that are executed.

Comment (pcnfernando) [reply compiler/ballerina-lang/src/main/java/io/ballerina/projects/ModuleContext.java:438]: No my question was whether "we"; in our context have any scenario that needed diagnostics to be checked after a unhandled exception

Comment (lochana-chathura) [reply compiler/ballerina-lang/src/main/java/io/ballerina/projects/ModuleContext.java:438]: Resolving as we discussed this offline.

**Review by hasithaa — DISMISSED** [review]

**Comment by hasithaa** [comment]
Shall we change the base to master?

**Review by hasithaa — APPROVED** [review]

**Pull request merged**